File: Term_Paper_1st_semester/Repeated_Number_Finder.py
N = int(input("Enter a number between 1 and 999: "))

seen_numbers = set()

while True:
    if N in seen_numbers:
        print(N)
        print("First repeated number:", N)
        break

    print(N, end=" → ")
    seen_numbers.add(N)
    digits = [int(d) for d in str(N)]
    N = sum(d * d for d in digits)

File: Term_Paper_1st_semester/Repeated_Number_Finder.cpp
#include <iostream>
#include <set>
using namespace std;

int main() {
    int N;
    cout << "Enter a number between 1 and 999: ";
    cin >> N;

    set<int> seenNumbers;
    cout << N;

    while (true) {
        if (seenNumbers.count(N)) {
            cout << endl << "First repeated number: " << N << endl;
            break;
        }

        seenNumbers.insert(N);

        int sum = 0, temp = N;
        while (temp > 0) {
            int digit = temp % 10;
            sum += digit * digit;
            temp /= 10;
        }

        N = sum;
        cout << " -> " << N;
    }

    return 0;
}
